IotHubConsumer/ReceiverConsumer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Consumer;
+using Microsoft.Extensions.Configuration;
+
+namespace IotHubConsumer
+{
+    public class ReceiverConsumer
+    {
+        private readonly IConfiguration configuration;
+        public HashSet<TestElement> MessagesWithRetainSet { get; set; } = new HashSet<TestElement>();
```

Comment:
Could you not return this list ReceiveMessagesFromDeviceAsync method, instead of making it a public variable?

Reply:
I did not want to modify the behaviour of the method, in case we use in other places except for the unit test.

Reply:
I think in that case, it would make sense to try to make two public functions, one that returns the list when after cancelation and a second without, with a shared private method that contains the shared logic.

I feel the current implementation isn't very intuitive to use.

Reply:
Tried to do some refactoring to increase readability. Once we add the offset, we should get rid of that list without changing the method.